### PR TITLE
transforms: update documentation

### PR DESCRIPTION
**Pull request by rockwotj**
We updated the APIs, so document the new SDKs.

Also add schema registry to the reference documentation.

### CODE REVIEW

**Comment by Deflaimun** [comment]
@rockwotj are there any updates for the rpk commands associated with transforms?

**Comment by rockwotj** [comment]
Let me double check

**Comment by rockwotj** [comment]
@Deflaimun there are updates yes, do you generate the docs for RPK or is it a manual process to update those?

**Comment by rockwotj** [comment]
It also looks like some public metrics for transforms are missing too - are those manual updates then? I can add those

**Comment by Deflaimun** [comment]
@rockwotj we run a script to get the latest updates from the available version. I'll update it right away

**Comment by Deflaimun** [comment]
Metrics are fully manual updates. @kbatuigas

**Comment by rockwotj** [comment]
Thanks @Deflaimun, I will update the metrics then

**Comment by Deflaimun** [comment]
rpk update PR
https://github.com/redpanda-data/docs/pull/238

File: modules/reference/pages/data-transform-api.adoc [hunk 1]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+      // by the broker.
+      Write(Record) error
+}
+```
+
+`RecordWriter` is an interface for writing transformed records to the destination topic.
+
+== Schema Registry Client
+
+Package `sr` is a Schema Registry client for usage within Redpanda Data Transforms.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
Package `sr` is a Schema Registry client for usage within Redpanda data transforms.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 2]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+Data transforms support reading schemas and writing schemas to the schema registry within Redpanda. 
+
+Once a schema is obtained, a record maybe decoded using the appropriate library. Using the TinyGo compiler and runtime, the following libraries for Apache Avro and Protocol Buffers have been known to work.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
After a schema is obtained, a record may be decoded using the appropriate library. Using the TinyGo compiler and runtime, the following libraries for Apache Avro and Protocol Buffers are known to work:
```

File: modules/reference/pages/data-transform-api.adoc [hunk 3]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+==== ExtractID
+
+```go
+func ExtractID(b []byte) (int, error)
+```
+
+Extract the ID from the header of a schema registry encoded value.
+
+Returns ErrBadHeader if the array is missing the leading magic byte or is too small.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
Returns `ErrBadHeader` if the array is missing the leading magic byte or is too small.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 4]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+==== MaxCacheEntries
+
+```go
+func MaxCacheEntries(size int) ClientOpt
+```
+
+`MaxCacheEntries` configures how many entries to cache within the client.
+
+By default the cache is unbounded, use 0 to disable the cache.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
By default the cache is unbounded. Use 0 to disable the cache. 
```

File: modules/reference/pages/data-transform-api.adoc [hunk 5]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+```go
+type Reference struct {
+	Name    string
+	Subject string
+	Version int
+}
+```
+
+`SchemaReference` is a way for a one schema to reference another. The details for how referencing is done are type specific; for example, JSON objects that use the key "$ref" can refer to another schema via URL. For more details on references, see the following: xref:manage:schema-reg/schema-reg-api.adoc#reference-a-schema[Reference a schema]
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`SchemaReference` is a way for one schema to reference another schema. The details for how referencing is done are type specific; for example, JSON objects that use the key "$ref" can refer to another schema via URL. See xref:manage:schema-reg/schema-reg-api.adoc#reference-a-schema[Reference a schema]
```

File: modules/reference/pages/data-transform-api.adoc [hunk 6]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+```go
+type Schema struct {
+	Schema     string
+	Type       SchemaType
+	References []Reference
+}
+```
+
+`Schema` is a schema that can be registered within schema registry
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`Schema` is a schema that can be registered within the Schema Registry. 
```

File: modules/reference/pages/data-transform-api.adoc [hunk 7]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+	// If an equivalent schema already exists globally, that schema ID will be reused.
+	// If an equivalent schema already exists within that subject, this will be a noop and the
+	// existing schema will be returned.
+	CreateSchema(subject string, schema Schema) (s *SubjectSchema, err error)
+}
+```
+
+`SchemaRegistryClient` is a client for interacting with the schema registry within Redpanda.
+
+The client provides caching out of the box, which can be configured with options.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`SchemaRegistryClient` is a client for interacting with Redpanda's Schema Registry.

The client provides caching out of the box, which can be configured with options.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 8]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+`NewClient` creates a new <<schemaregistryclient, `SchemaRegistryClient`>> with the specified options applied.
+
+==== SchemaType
+
+```go
+type SchemaType int
+```
+
+`SchemaType` is an enum for the different types of schemas that can be stored in schema registry.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`SchemaType` is an enum for the different types of schemas that can be stored in the Schema Registry.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 9]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+}
+```
+
+`RecordWriter` is an interface for writing transformed records to the destination topic.
+
+== Schema Registry Client
+
+Package `sr` is a Schema Registry client for usage within Redpanda Data Transforms.
+
+Data transforms support reading schemas and writing schemas to the schema registry within Redpanda.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
Data transforms support reading schemas and writing schemas to the Redpanda's Schema Registry. 
```

File: modules/reference/pages/data-transform-api.adoc [hunk 10]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+```go
+type Serde[T any] struct {
+	// contains filtered or unexported fields
+}
+```
+
+`Serde` encodes and decodes values according to the schema registry wire format. A `Serde` itself does not perform schema auto-discovery and type auto-decoding. To aid in strong typing and validated encoding/decoding, you must register IDs and values to how to encode or decode them.
+
+To use a `Serde` for encoding, you must pre-register schema ids and values you will encode, and then you can use the encode functions. The latest registered ID that supports encoding will be used to encode.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
To use a `Serde` for encoding, you must first preregister the schema IDs and values that you will encode. The latest registered ID that supports encoding is used to encode.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 11]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+To use a `Serde` for decoding, you can either pre-register schema ids and values you will consume, or you can discover the schema every time you receive an `ErrNotRegistered` error from decode.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
To use a `Serde` for decoding, you can either preregister the schema IDs and values that you will consume, or you can discover the schema every time you receive an `ErrNotRegistered` error from decode.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 12]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+==== (*Serde[T]) AppendEncode
+
+```go
+func (s *Serde[T]) AppendEncode(b []byte, v T) ([]byte, error)
+```
+
+`AppendEncode` appends an encoded value to b according to the schema registry wire format and returns it. If <<encodefn,`EncodeFn`>> was not used, this returns `ErrNotRegistered`.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`AppendEncode` appends an encoded value to `b` according to the schema registry wire format and returns it. If <<encodefn,`EncodeFn`>> was not used, this returns `ErrNotRegistered`.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 13]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+==== (*Serde[T]) Decode
+
+```go
+func (s *Serde[T]) Decode(b []byte, v T) error
+```
+
+`Decode` decodes b into v. If <<decodefn, `DecodeFn`>> option was not used, this returns `ErrNotRegistered`.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`Decode` decodes `b` into `v`. If the <<decodefn, `DecodeFn`>> option was not used, this returns `ErrNotRegistered`.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 14]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+Serde does not handle references in schemas; it is up to you to register the full decode function for any top-level ID, regardless of how many other schemas are referenced in top-level ID.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
Serde does not handle references in schemas. You must register the full decode function for any top-level ID, regardless of how many other schemas are referenced in the top-level ID.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 15]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+      // When writing a record, only the key, value and headers are
+      // used other information like the timestamp will be overridden
+      // by the broker.
+      Write(Record) error
+}
+```
+
+`RecordWriter` is an interface for writing transformed records to the destination topic.
+
+== Schema Registry Client
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
== Schema Registry client
```

File: modules/reference/pages/data-transform-api.adoc [hunk 16]
```diff
@@ -4,9 +4,13 @@
 
 API reference documentation for Redpanda data transforms.
 
-== Data transforms functions
+== Data Transforms
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
== Data transforms
```

File: modules/reference/pages/data-transform-api.adoc [hunk 17]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+Serde does not handle references in schemas; it is up to you to register the full decode function for any top-level ID, regardless of how many other schemas are referenced in top-level ID.
+
+==== (*Serde[T]) Encode
+
+```go
+func (s *Serde[T]) Encode(v T) ([]byte, error)
+```
+
+`Encode` encodes a value according to the schema registry wire format and returns it. If <<encodefn, `EncodeFn`>> was not used, this returns `ErrNotRegistered`.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`Encode` encodes a value according to the Schema Registry wire format and returns it. If <<encodefn, `EncodeFn`>> was not used, this returns `ErrNotRegistered`.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 18]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+`MustAppendEncode` returns the value of <<serdet-append-encode, `AppendEncode`>>, panicking on error. This is a shortcut for if your encode function cannot error.
+
+==== (*Serde[T]) MustEncode
+
+```go
+func (s *Serde[T]) MustEncode(v T) []byte
+```
+
+`MustEncode` returns the value of <<serdet-encode, `Encode`>>, panicking on error. This is a shortcut for if your encode function cannot error.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`MustEncode` returns the value of <<serdet-encode, `Encode`>>, panicking on error. This is a shortcut for when your encode function cannot error.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 19]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+==== (*Serde[T]) Register
+
+```go
+func (s *Serde[T]) Register(id int, opts ...SerdeOpt[T])
+```
+
+`Register` registers a schema ID and the value it corresponds to, as well as the encoding or decoding functions. You need to register functions depending on whether you are only encoding, only decoding, or both.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`Register` registers a schema ID and the value it corresponds to, as well as the encoding or decoding functions. Register functions depending on whether you are only encoding, only decoding, or both.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 20]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+type SubjectSchema struct {
+	Schema
+
+	Subject string
+	Version int
+	ID      int
+}
+```
+
+`SchemaSubject` is a schema along with the subject, version and ID of the schema
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`SchemaSubject` is a schema along with the subject, version, and ID of the schema. 
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 21]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_transform_execution_errors
+
+Data transform invocation errors
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
Data transform invocation errors.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 22]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_wasm_engine_cpu_seconds_total
+
+Total CPU Time (in seconds) spent inside a WebAssembly function
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
Total CPU time (in seconds) spent inside a WebAssembly function.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 23]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_wasm_engine_memory_usage
+
+Amount of memory usage for a WebAssembly function
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
Amount of memory usage for a WebAssembly function.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 24]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_wasm_engine_max_memory
+
+Max amount of memory for a WebAssembly function
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
Maximum amount of memory for a WebAssembly function.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 25]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_wasm_binary_executable_memory_usage
+
+The amount of executable memory usage for WebAssembly binaries
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
The amount of executable memory usage for WebAssembly binaries.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 26]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Type*: gauge
+
+---
+
+=== redpanda_transform_read_bytes
+
+A counter for all the bytes that has been input to a transform
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
A counter for all the bytes that has been input to a transform.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 27]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_transform_write_bytes
+
+A counter for all the bytes that has been output from a transform
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
A counter for all the bytes that has been output from a transform.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 28]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_transform_processor_lag
+
+The amount of pending records on the input topic that have yet to be processed by the transform
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
The amount of pending records on the input topic that have yet to be processed by the transform.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 29]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_transform_failures
+
+A counter for each time that a processor encounters a failure
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
A counter for each time that a processor encounters a failure.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 30]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_transform_state
+
+The number of transform processors in a specific state (running, inactive, errored)
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
The number of transform processors in a specific state (running, inactive, errored).
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 31]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
 
 ---
 
+[[data_transform_metrics]]
+== Data Transforms metrics
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
== Data transforms metrics
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 32]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
 
 ---
 
+[[data_transform_metrics]]
+== Data transforms metrics
+
+=== redpanda_transform_execution_latency_sec
+
+A histogram of the latency in seconds of transforming a single record.
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
A histogram of the latency of transforming a single record, in seconds.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 33]
```diff
@@ -746,6 +746,138 @@ Indicates whether a given resource has at least one valid installed TLS certific
...
+
+*Labels*:
+
+* `function_name`
+
+---
+
+=== redpanda_wasm_engine_cpu_seconds_total
+
+Total CPU time (in seconds) spent inside a WebAssembly function.
```
Comment (micheleRP) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
Total CPU time spent inside a WebAssembly function, in seconds.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 34]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
 
 The `WriteEvent` type contains information about a record that was written.
 
+==== RecordWriter
+
+```go
+type RecordWriter interface {
+      // Write writes a record to the output topic.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc:100]:
```suggestion
      // Writes a record to the output topic.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 35]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+}
+```
+
+`Schema` is a schema that can be registered within the Schema Registry. 
+
+==== SchemaRegistryClient
+
+```go
+type SchemaRegistryClient interface {
+	// LookupSchemaById looks up a schema via it's global ID.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
	// LookupSchemaById looks up a schema via its global ID.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 36]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+}
+```
+
+`RecordWriter` is an interface for writing transformed records to the destination topic.
+
+== Schema Registry client
+
+Package `sr` is a Schema Registry client for usage within Redpanda data transforms.
+
+Data transforms support reading schemas and writing schemas to the Redpanda's Schema Registry.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
Data transforms support reading schemas and writing schemas to Redpanda's Schema Registry. 
```

File: modules/reference/pages/data-transform-api.adoc [hunk 37]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+=== Schema Registry functions
+
+==== ExtractID
+
+```go
+func ExtractID(b []byte) (int, error)
+```
+
+Extract the ID from the header of a schema registry encoded value.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
Extract the ID from the header of a Schema Registry encoded value.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 38]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+```go
+type Reference struct {
+	Name    string
+	Subject string
+	Version int
+}
+```
+
+`SchemaReference` is a way for one schema to reference another schema. The details for how referencing is done are type specific; for example, JSON objects that use the key "$ref" can refer to another schema via URL. See xref:manage:schema-reg/schema-reg-api.adoc#reference-a-schema[Reference a schema]
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`SchemaReference` is a way for one schema to reference another schema. The details for how referencing is done are type specific; for example, JSON objects that use the key "$ref" can refer to another schema via URL. See xref:manage:schema-reg/schema-reg-api.adoc#reference-a-schema[Reference a schema].
```

File: modules/reference/pages/data-transform-api.adoc [hunk 39]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+==== Serde
+
+```go
+type Serde[T any] struct {
+	// contains filtered or unexported fields
+}
+```
+
+`Serde` encodes and decodes values according to the schema registry wire format. A `Serde` itself does not perform schema auto-discovery and type auto-decoding. To aid in strong typing and validated encoding/decoding, you must register IDs and values to how to encode or decode them.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`Serde` encodes and decodes values according to the Schema Registry wire format. A `Serde` itself does not perform schema auto-discovery and type auto-decoding. To aid in strong typing and validated encoding/decoding, you must register IDs and values.
```

File: modules/reference/pages/data-transform-api.adoc [hunk 40]
```diff
@@ -95,7 +93,274 @@ type WriteEvent interface {
...
+
+`Encode` encodes a value according to the Schema Registry wire format and returns it. If <<encodefn, `EncodeFn`>> was not used, this returns `ErrNotRegistered`.
+
+==== (*Serde[T]) MustAppendEncode
+
+```go
+func (s *Serde[T]) MustAppendEncode(b []byte, v T) []byte
+```
+
+`MustAppendEncode` returns the value of <<serdet-append-encode, `AppendEncode`>>, panicking on error. This is a shortcut for if your encode function cannot error.
```
Comment (micheleRP) [inline modules/reference/pages/data-transform-api.adoc]:
```suggestion
`MustAppendEncode` returns the value of <<serdet-append-encode, `AppendEncode`>>, panicking on error. This is a shortcut for when your encode function cannot error.
```

File: modules/reference/pages/data-transform-api.adoc (hunk 34 above)
Comment (rockwotj) [reply modules/reference/pages/data-transform-api.adoc:100]: This is a standard golang pattern, as awkward as it is: https://go.dev/blog/godoc

Quote:

> Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.

**Review by micheleRP — APPROVED** [review]
looks great, thank you!

**Pull request merged**